pkg/utils/test/matchers/managedresource_test.go

```diff
@@ -0,0 +1,166 @@
+// Copyright 2024 SAP SE or an SAP affiliate company. All rights reserved. This file is licensed under the Apache Software License, v. 2 except as noted otherwise in the LICENSE file
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package matchers_test
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"github.com/gardener/etcd-druid/pkg/client/kubernetes"
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	"github.com/onsi/gomega/types"
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/utils/ptr"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	fakeclient "sigs.k8s.io/controller-runtime/pkg/client/fake"
+
+	resourcesv1alpha1 "github.com/gardener/gardener/pkg/apis/resources/v1alpha1"
+	. "github.com/gardener/gardener/pkg/utils/test/matchers"
+	testruntime "github.com/gardener/gardener/pkg/utils/test/runtime"
+)
+
+var _ = Describe("ManagedResource Object Matcher", func() {
+	var (
+		fakeClient     client.Client
+		containsObject func(client.Object) types.GomegaMatcher
+	)
+
+	BeforeEach(func() {
+		fakeClient = fakeclient.NewClientBuilder().WithScheme(kubernetes.Scheme).Build()
+		containsObject = NewManagedResourceObjectMatcher(fakeClient)
+	})
+
+	Context("without managed resource", func() {
+		It("should not find an object", func() {
+			Expect(nil).ToNot(containsObject(&corev1.Secret{}))
```

Comment:
```suggestion
			Expect(nil).NotTo(containsObject(&corev1.Secret{}))
```
and other places for consistency